### ISSUE TITLE
 __init__() missing 4 required positional arguments: 'casting', 'from_', 'to', and 'i'

### ISSUE THREAD

**Issue by FirasKedidi** [open-mmlab/mmdetection#5026]
Hello ,
I am trying to use a pretrained mask rcnn model on my custom dataset and I faced this error.Please note that I am following the getting started colab tutorial and the documentation. I added the data classes and the model number of classes only.I am using colab
![image](https://user-images.githubusercontent.com/58661767/115358801-f070aa00-a1b5-11eb-9e45-d0f624918a54.png)

**Comment by FirasKedidi** [comment]
`cfg = Config.fromfile('/content/mmdetection/configs/mask_rcnn/mask_rcnn_r50_caffe_fpn_mstrain-poly_3x_coco.py')
cfg.dataset_type = 'CocoDataset'
cfg.classes = ("article","flag","footer")
# number of classes

cfg.data.train.img_prefix = '/content/drive/MyDrive/training_dataset_1/'
cfg.data.train.classes = cfg.classes
cfg.data.train.ann_file = '/content/drive/MyDrive/training_dataset_1/training_data.json'
cfg.data.train.type = 'CocoDataset'
cfg.data.val.img_prefix = '/content/vgg/'
cfg.data.val.classes = cfg.classes
cfg.data.val.ann_file = '/content/vgg/arabic_newspaper_coco.json'
cfg.data.val.type = 'CocoDataset'
cfg.data.test.img_prefix = '/content/drive/MyDrive/validation (1)'
cfg.data.test.classes = cfg.classes
cfg.data.test.ann_file = '/content/drive/MyDrive/validation (1)/validation_data.json'
cfg.data.test.type = 'CocoDataset'
cfg.model.roi_head.bbox_head.num_classes=3
cfg.model.roi_head.mask_head.num_classes=3
cfg.load_from = 'checkpoints/mask_rcnn_r50_caffe_fpn_mstrain-poly_3x_coco_bbox_mAP-0.408__segm_mAP-0.37_20200504_163245-42aa3d00.pth'
cfg.work_dir = "/content/output"
cfg.runner.max_epochs = 2
cfg.total_epochs = 2
cfg.optimizer.lr = 0.02 / 8
cfg.lr_config.warmup = None
cfg.log_config.interval = 500
# Change the evaluation metric since we use customized dataset.
cfg.evaluation.metric = 'mAP'
# We can set the evaluation interval to reduce the evaluation times
cfg.evaluation.interval = 3
# We can set the checkpoint saving interval to reduce the storage cost
cfg.checkpoint_config.interval = 3
# Set seed thus the results are more reproducible
cfg.seed = 0
set_random_seed(0, deterministic=False)
cfg.gpu_ids = range(1)


print(f'Config:\n{cfg.pretty_text}')
from mmdet.datasets import build_dataset
from mmdet.models import build_detector
from mmdet.apis import train_detector


# Build dataset
datasets = [build_dataset(cfg.data.train)]

# Build the detector
model = build_detector(
    cfg.model, train_cfg=cfg.get('train_cfg'), test_cfg=cfg.get('test_cfg'))
# Add an attribute for visualization convenience
model.CLASSES = datasets[0].CLASSES

# Create work_dir
mmcv.mkdir_or_exist(osp.abspath(cfg.work_dir))
train_detector(model, datasets, cfg, distributed=False, validate=True)`

**Comment by FirasKedidi** [comment]
the error is in the level of train_detector(model, datasets, cfg, distributed=False, validate=True)

**Comment by hhaAndroid** [comment]
This error is very strange, can you directly use `train.py` to train？

**Comment by FirasKedidi** [comment]
> This error is very strange, can you directly use `train.py` to train？

No I've used build_detector(
cfg.model, train_cfg=cfg.get('train_cfg'), test_cfg=cfg.get('test_cfg')).
Like in the colab tutorial

**Comment by FirasKedidi** [comment]
> This error is very strange, can you directly use `train.py` to train？

thank you for your reply.
I've tried cascade mask rcnn another model and I didn't face the error

**Comment by apanand14** [comment]
Hey @FirasKedidi @hhaAndroid , I'm facing same issue currently for some json files. Can you guide me what should I do to avoid this issue??
